src/logic/unique.lean

```diff
@@ -137,22 +137,35 @@ lemma unique_iff_subsingleton_and_nonempty (α : Sort u) :
 ⟨λ ⟨u⟩, by split; exactI infer_instance,
  λ ⟨hs, hn⟩, ⟨by { resetI, inhabit α, exact unique.mk' α }⟩⟩
 
-@[simp] lemma pi.default_def {β : Π a : α, Sort v} [Π a, inhabited (β a)] :
+@[simp] lemma pi.default_def {β : α → Sort v} [Π a, inhabited (β a)] :
   @default (Π a, β a) _ = λ a : α, @default (β a) _ := rfl
 
-lemma pi.default_apply {β : Π a : α, Sort v} [Π a, inhabited (β a)] (a : α) :
+lemma pi.default_apply {β : α → Sort v} [Π a, inhabited (β a)] (a : α) :
   @default (Π a, β a) _ a = default := rfl
 
-instance pi.unique {β : Π a : α, Sort v} [Π a, unique (β a)] : unique (Π a, β a) :=
+instance pi.unique {β : α → Sort v} [Π a, unique (β a)] : unique (Π a, β a) :=
 { uniq := λ f, funext $ λ x, unique.eq_default _,
   .. pi.inhabited α }
 
 /-- There is a unique function on an empty domain. -/
-instance pi.unique_of_is_empty [is_empty α] (β : Π a : α, Sort v) :
+instance pi.unique_of_is_empty [is_empty α] (β : α → Sort v) :
   unique (Π a, β a) :=
 { default := is_empty_elim,
   uniq := λ f, funext is_empty_elim }
 
+lemma congr_arg_heq_of_subsingleton [subsingleton α] {β : α → Sort v}
+  (f : Π a, β a) (i j : α) : f i == f j := by rw subsingleton.elim i j
+
+lemma congr_arg_of_subsingleton [subsingleton α]
+  (f : α → β) (i j : α) : f i = f j := by rw subsingleton.elim i j
```

Comment:
Longer, but makes the naming more apparent:
```suggestion
  (f : α → β) (i j : α) : f i = f j := congr_arg f $ subsingleton.elim i j
```
It's not immediately clear to me why this lemma is more useful than say `congr_arg_zero_add : f (0 + n) = f n`, and therefore whether we should actually have it.

Reply:
Indeed I should probably remove it; it's already removed in the homotopy group PR.